### PR TITLE
Feature/coins list

### PR DESCRIPTION
**Pull request by behzadiv**

### CODE REVIEW

File: src/components/coinsList/content/AddToFavorite.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { FavoriteActionHandler } from "@redux/actions/favoriteAction";
+import { StarOutlined, StarFilled } from "@ant-design/icons";
+
+export const AddToFavoriteSection = ({ props }: any) => {
+  const coin = props;
+  const dispatch = useDispatch();
+  const favoriteAction = (coin: {
+    iconUrl: string;
+    name: string;
+    price: string;
```
Comment (alirezatahani0) [inline src/components/coinsList/content/AddToFavorite.tsx]: move to types file

File: src/components/coinsList/content/AddToFavorite.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { FavBtn } from "../styles/addToFavorite_style";
+import { FavoriteActionHandler } from "@redux/actions/favoriteAction";
+import { StarOutlined, StarFilled } from "@ant-design/icons";
+
+export const AddToFavoriteSection = ({ props }: any) => {
+  const coin = props;
```
Comment (alirezatahani0) [inline src/components/coinsList/content/AddToFavorite.tsx]: destruct from component definition

File: src/components/coinsList/content/AddToFavorite.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  const dispatch = useDispatch();
+  const favoriteAction = (coin: {
+    iconUrl: string;
+    name: string;
+    price: string;
+  }) => {
+    dispatch(FavoriteActionHandler(coin));
+  };
+  const { favoriteList } = useSelector((state: any) => state.FavoriteReducer);
```
Comment (alirezatahani0) [inline src/components/coinsList/content/AddToFavorite.tsx]: order of usage 

1. custom hooks 
2. states
3. functions
4. hooks

File: src/components/coinsList/content/CoinChangePrice.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+} from "../styles/coinPriceChange_style";
+
+export const CoinPriceChangeSection = (props: any) => {
+  const priceChange = props.props;
+  return priceChange === null ? (
+    <CoinNegetiveChange>0</CoinNegetiveChange>
+  ) : priceChange.includes("-") ? (
+    <CoinNegetiveChange>{priceChange}</CoinNegetiveChange>
+  ) : (
+    <CoinPositiveChange>+{priceChange}</CoinPositiveChange>
```
Comment (alirezatahani0) [inline src/components/coinsList/content/CoinChangePrice.tsx]: clean this lines 

if (priceChange === null) return {something}

File: src/components/coinsList/content/CoinListContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      ) : (
+        data &&
+        data.map(
+          (coin: {
+            iconUrl: string;
+            name: string;
+            price: string;
+            change: string;
+            uuid: string;
+          }) => {
```
Comment (alirezatahani0) [inline src/components/coinsList/content/CoinListContainer.tsx]: move interfaces to types file

File: src/components/coinsList/content/CoinListContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            return (
+              <CoinContent key={coin.uuid}>
+                <CoinDesc>
+                  <img src={coin.iconUrl} style={{ width: 40 }} />
+                  <CoinName> {coin.name}</CoinName>
+                </CoinDesc>
+                <CoinPriceSection props={coin.price} />
```
Comment (alirezatahani0) [inline src/components/coinsList/content/CoinListContainer.tsx]: pass props as specific name

File: src/components/coinsList/content/CoinListContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            iconUrl: string;
+            name: string;
+            price: string;
+            change: string;
+            uuid: string;
+          }) => {
+            return (
+              <CoinContent key={coin.uuid}>
+                <CoinDesc>
+                  <img src={coin.iconUrl} style={{ width: 40 }} />
```
Comment (alirezatahani0) [inline src/components/coinsList/content/CoinListContainer.tsx]: destruct values from coin to simpler usage

File: src/components/coinsList/content/CoinPrice.tsx [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import * as React from "react";
+import { formatPrice } from "../utils/formatPrice";
```
Comment (alirezatahani0) [inline src/components/coinsList/content/CoinPrice.tsx]: move numberToPrice to general utils

File: src/components/coinsList/styles/coinListContainer_style.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import styled from "styled-components";
+
+export const CoinListContainer = styled.div(() => ({
+  marginTop: "1rem",
+}));
+export const CoinContent = styled.div(({}) => ({
+  display: "flex",
+  gap: "2rem",
+  marginBottom: "1rem",
+  color: "white",
```
Comment (alirezatahani0) [inline src/components/coinsList/styles/coinListContainer_style.ts]: use from theme

File: src/components/coinsList/styles/coinPrice_style.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+export const CoinPrice = styled.p<CoinPriceProps>(({ theme }) => ({
+  fontSize: theme.typography.body1.fontSize,
+  borderRadius: 4,
+  textAlign: "start",
+  padding: 2,
+  width: 90,
+  flex: 1,
+}));
+
+export interface CoinPriceProps {
```
Comment (alirezatahani0) [inline src/components/coinsList/styles/coinPrice_style.ts]: move to types file

File: src/components/coinsList/utils/formatPrice.ts [hunk 11]
```diff
@@ -0,0 +1,5 @@
+export const formatPrice = (price: number) => {
```
Comment (alirezatahani0) [inline src/components/coinsList/utils/formatPrice.ts]: change name to numberToPrice

**Review by alirezatahani0 — CHANGES_REQUESTED** [review]

File: src/components/coinsList/content/AddToFavorite.tsx (hunk 1 above)
Comment (behzadiv) [reply src/components/coinsList/content/AddToFavorite.tsx]: done!

File: src/components/coinsList/content/AddToFavorite.tsx (hunk 2 above)
Comment (behzadiv) [reply src/components/coinsList/content/AddToFavorite.tsx]: done

File: src/components/coinsList/content/CoinList.tsx [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+  return (
+    <CoinListContainer>
+      {loading ? (
+        <Spin />
+      ) : (
+        data &&
+        data.map((coin: CoinProps) => {
+          const { change, iconUrl, name, price, uuid } = coin;
+          return (
+            <CoinContent key={uuid}>
```
Comment (alirezatahani0) [inline src/components/coinsList/content/CoinList.tsx]: create a component from it 
Coin, CoinItem

File: src/components/coinsList/content/AddToFavorite.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { AddToFavoriteProps } from "../types/addToFavorite_types";
+import { FavBtn } from "../styles/addToFavorite_style";
+import { FavoriteActionHandler } from "@redux/actions/favoriteAction";
+import { StarOutlined, StarFilled } from "@ant-design/icons";
+
+export const AddToFavoriteSection: React.FC<AddToFavoriteProps> = ({coin}) => {
+  const { favoriteList } = useSelector((state: any) => state.FavoriteReducer);
```
Comment (alirezatahani0) [inline src/components/coinsList/content/AddToFavorite.tsx]: dont use any

File: src/components/coinsList/styles/coinPriceChange_style.ts [hunk 14]
```diff
@@ -0,0 +1,17 @@
+import styled from "styled-components";
+
+export const CoinNegetiveChange = styled.p(({ theme }) => ({
```
Comment (alirezatahani0) [inline src/components/coinsList/styles/coinPriceChange_style.ts]: merge it with positive

File: src/components/coinsList/content/CoinChangePrice.tsx [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+import {
+  CoinNegetiveChange,
+  CoinPositiveChange,
+} from "../styles/coinPriceChange_style";
+
+export const CoinPriceChangeSection: React.FC<CoinPriceChangeProps> = ({
+  priceChange,
+}) => {
+  if (priceChange === null) return <CoinNegetiveChange>0</CoinNegetiveChange>;
+  return priceChange.includes("-") ? (
```
Comment (alirezatahani0) [inline src/components/coinsList/content/CoinChangePrice.tsx]: change to use JS Number functions

File: src/components/coinsList/content/CoinPrice.tsx [hunk 16]
```diff
@@ -0,0 +1,10 @@
+import * as React from "react";
+import { CoinPriceSectionProps } from "../types/coinPrice_types";
+import { formatPrice } from "../../../utils/numberToPrice";
```
Comment (alirezatahani0) [inline src/components/coinsList/content/CoinPrice.tsx]: add path to tsConfig

File: src/modules/getCoinsData/content/getCoinsData.tsx [hunk 17]
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from "react";
+import useFetch from "../../../hooks/useFetch";
```
Comment (alirezatahani0) [inline src/modules/getCoinsData/content/getCoinsData.tsx:2]: add to tsConfig Path

**Review by alirezatahani0 — CHANGES_REQUESTED** [review]

**Review by alirezatahani0 — APPROVED** [review]

**Pull request merged**